### ISSUE TITLE
StackOverflowError when comparing ZonedDateTime and DateTimes

### ISSUE THREAD

**Issue by joshbode** [JuliaTime/TimeZones.jl#52]
Though comparison of a `ZonedDateTime` with a `DateTime` might not be particularly valid/useful (which timezone should be used for the `DateTime`?), comparisons such as `>, >=, <=, <` are generating a non-intuitive `StackOverflowError`:

```julia
julia> using TimeZones

julia> x = ZonedDateTime(now(), localzone())
2017-03-20T13:01:12.176+11:00

julia> x == now()
false

julia> x != now()
true

julia> x > now()
------ StackOverflowError -------------- Stacktrace (most recent call last)

 [1] — isless(::TimeZones.ZonedDateTime, ::DateTime) at types.jl:241 (repeats 80000 times)

StackOverflowError:
```

**Comment by omus** [comment]
Looks like we need some promotion rules to make it explicit that `promote_type(ZonedDateTime, DateTime)` doesn't work.

**Comment by omus** [comment]
Fixed by #54

**Comment by joshbode** [comment]
Thanks!
